src/Dialect/ONNX/MLIRDialectBuilder.hpp

```diff
@@ -166,21 +166,28 @@ struct VectorBuilder final : DialectBuilder {
   VectorBuilder(OpBuilder &b, Location loc) : DialectBuilder(b, loc) {}
   VectorBuilder(DialectBuilder &db) : DialectBuilder(db) {}
 
+  // Get the machine SIMD vector length for the given elementary type.
+  // This can help guide certain optimizations.
+  int64_t getMachineVectorLength(const Type &elementType);
+  int64_t getMachineVectorLength(const VectorType &vecType);
+  int64_t getMachineVectorLength(Value vecValue);
```

Comment:
I think all these new member functions should be declared`const`, they aren't intended to change the state of an object of this class type.

Reply:
good point